src/geometry/dual_quaternion.rs

```diff
@@ -77,8 +84,147 @@ where
     /// relative_eq!(dq.real.norm(), 1.0);
     /// ```
     #[inline]
-    pub fn normalize_mut(&mut self) {
-        *self = self.normalize();
+    pub fn normalize_mut(&mut self) -> N {
```

Comment:
It's idiomatic for `normalize_mut` functions to return the norm of the value prior to normalization, as it's a necessary intermediate step to normalize the value anyway, and it's very often useful alongside the normalized value. For example, in a function where you're performing a raycast, it can be useful to normalize the ray to produce a unit direction vector, while also getting the norm of the ray to scale the time-of-impact factor back to being a parameter along the ray rather than a scaling factor for the direction vector. The [implementation of normalize_mut for vectors](https://docs.rs/nalgebra/0.24.0/nalgebra/base/struct.Matrix.html#method.normalize_mut) works this way too.

It's probably not *as* useful for a dual quaternion, as unit dual quaternions are much more commonly used... but it provides consistency, and I've found a use for it in my project, where I encode similarities instead of isometries by encoding the scale factor in the norm of the real part.